src/metabase/util/malli/humanize.cljc

```diff
@@ -0,0 +1,71 @@
+(ns metabase.util.malli.humanize
+  (:require
+   [malli.error :as me]))
+
+(defn- resolve-error
+  "This is the same behavior as what [[malli.error/humanize]] does to resolve errors."
+  [explanation error]
+  (me/-resolve-direct-error explanation error {:wrap :message, :resolve me/-resolve-direct-error}))
+
+(defn- flatten-error
+  "Given a `[path message]` pair like
+
+    [[2] \"some error\"]
+
+  return a flattened error message like
+
+    [nil nil \"some error\"]"
+  [[path message]]
+  (if (empty? path)
+    message
+    (recur
+     [(butlast path)
+      (if (integer? (last path))
+        (me/-push [] (last path) message nil)
+        {(last path) message})])))
```

Comment:
I think `loop`ing over `(reverse path)` would be easier to read (and somewhat faster). If you want it to look like processing a stack, I suggest `loop`ing over `(vec path)` and using `peek` and `pop`.